dart/lib/src/sentry_stack_trace_factory.dart

```diff
@@ -24,19 +24,26 @@ class SentryStackTraceFactory {
     return parse(stackTrace).frames;
   }
 
-  SentryStackTrace parse(dynamic stackTrace) {
+  SentryStackTrace parse(dynamic stackTrace, {bool? removeSentryFrames}) {
     final parsed = _parseStackTrace(stackTrace);
     final frames = <SentryStackFrame>[];
     var onlyAsyncGap = true;
 
     for (var t = 0; t < parsed.traces.length; t += 1) {
       final trace = parsed.traces[t];
 
-      // NOTE: We want to keep the Sentry frames for crash detection
+      // NOTE: We want to keep the Sentry frames for SDK crash detection
       // this does not affect grouping since they're not marked as inApp
+      // only exception if there was no stack trace, we remove them
       for (final frame in trace.frames) {
-        final stackTraceFrame = encodeStackTraceFrame(frame);
+        var stackTraceFrame = encodeStackTraceFrame(frame);
+
         if (stackTraceFrame != null) {
+          if (removeSentryFrames == true &&
+              (stackTraceFrame.package == 'sentry' ||
+                  stackTraceFrame.package == 'sentry_flutter')) {
+            continue;
```

Comment:
it's not only limited to 'sentry' and 'sentry_flutter'

see here: https://github.com/getsentry/sentry-dart/blob/f754e867516d18d38272967acc03289a4ff83c0c/dart/lib/src/sentry_stack_trace_factory.dart#L18C1-L29C1

Reply:
In this case only those are relevant, as we just want to remove those frames that originate from creating the fallback stack-trace, no?